CorePayments/src/main/java/com/paypal/android/corepayments/ResourceLoader.kt

```diff
@@ -21,19 +21,22 @@ class ResourceLoader {
      * @param context Android context
      * @param resId ID of the resource that will be loaded
      */
-    suspend fun loadRawResource(context: Context, @RawRes resId: Int): String = withContext(Dispatchers.IO) {
-        try {
-            val resInputStream = context.resources.openRawResource(resId)
-            val resAsBytes = ByteArray(resInputStream.available())
-            resInputStream.read(resAsBytes)
-            resInputStream.close()
-            String(resAsBytes)
-        } catch (e: Resources.NotFoundException) {
-            val errorDescription = "Resource with id $resId not found."
-            throw PayPalSDKError(0, errorDescription, reason = e)
-        } catch (e: IOException) {
-            val errorDescription = "Error loading resource with id $resId."
-            throw PayPalSDKError(0, errorDescription, reason = e)
+    suspend fun loadRawResource(context: Context, @RawRes resId: Int): LoadRawResourceResult =
+        withContext(Dispatchers.IO) {
+            try {
+                val resInputStream = context.resources.openRawResource(resId)
+                val resAsBytes = ByteArray(resInputStream.available())
+                resInputStream.read(resAsBytes)
+                resInputStream.close()
+                LoadRawResourceResult.Success(String(resAsBytes))
+            } catch (e: Resources.NotFoundException) {
+                val errorDescription = "Resource with id $resId not found."
+                val resourceNotFoundError = PayPalSDKError(0, errorDescription, reason = e)
+                LoadRawResourceResult.Failure(resourceNotFoundError)
+            } catch (e: IOException) {
+                val errorDescription = "Error loading resource with id $resId."
+                val ioError = PayPalSDKError(0, errorDescription, reason = e)
+                LoadRawResourceResult.Failure(ioError)
+            }
```

Comment:
Could it throw other exceptions?

Reply:
That's a good question. I took a thorough tour of the docs and here are all the methods called:

- [Context#getResources()](https://developer.android.com/reference/android/content/Context#getResources()) does not throw
- [Resources#openRawResource()](https://developer.android.com/reference/android/content/res/Resources#openRawResource(int,%20android.util.TypedValue)) `throws Resources.NotFoundException`
- [InputStream#available()](https://developer.android.com/reference/java/io/InputStream#available()) `throws IOException`
- [ByteArray()](https://docs.oracle.com/cd/E14571_01/apirefs.1111/e12063/oracle/dss/util/xdo/common/io/ByteArray.html#ByteArray(byte[])) does not throw
- [InputStream#read()](https://developer.android.com/reference/java/io/InputStream#read(byte[])) `throws IOException, NullPointerException`
- [InputStream#close()](https://developer.android.com/reference/java/io/InputStream#close()) `throws IOException`

According to the docs `InputStream#read()` will only throw if `resAsBytes` is `null`, and through Kotlin we have guarantees that it is non-null. All the other possible exceptions we've caught.